kse/src/org/kse/gui/actions/VerifyCertificateAction.java

```diff
@@ -126,13 +152,125 @@ private void verifyChain(KeyStoreHistory keyStoreHistory, String alias)
 	private void verifyStatusOCSP(KeyStoreHistory keyStoreHistory, String alias)
 			throws KeyStoreException, NoSuchAlgorithmException, CertificateException, IOException,
 			InvalidAlgorithmParameterException, CertPathValidatorException, IllegalStateException, CryptoException {
-		if (verify("false", "true", true, keyStoreHistory,null)) {
+		if (verify("false", "true", true, keyStoreHistory, null)) {
+			JOptionPane.showMessageDialog(frame, res.getString("VerifyCertificateAction.OcspSuccessful.message"),
+					res.getString("VerifyCertificateAction.Verify.Title") + " " + alias,
+					JOptionPane.INFORMATION_MESSAGE);
+		}
+	}
+
+	private void verifyStatusOcspUrl(KeyStoreHistory keyStoreHistory, String alias, String ocspUrl)
+			throws OperatorCreationException, OCSPException, MalformedURLException, IOException, HeadlessException,
+			CertPathValidatorException, KeyStoreException, NoSuchAlgorithmException, CertificateException {
+
+		X509Certificate issuer = null;
+		KeyStore trustStore = getKeyStore(keyStoreHistory);
+		Enumeration<String> enumeration = trustStore.aliases();
+		while (enumeration.hasMoreElements()) {
+			String tempAlias = enumeration.nextElement();
+			X509Certificate cert = (X509Certificate) trustStore.getCertificate(tempAlias);
+			try {
+				certificateEval.verify(cert.getPublicKey());
+				issuer = cert;
+				break;
+			} catch (InvalidKeyException | CertificateException | NoSuchAlgorithmException | NoSuchProviderException
+					| SignatureException e) {
+				// ignore
+			}
+		}
```

Comment:
Maybe combine the Java cert path validation with the manual OCSP check?

Reply:
It's done.